cpp/src/arrow/util/logging.h

```diff
@@ -183,9 +186,6 @@ class ARROW_EXPORT ArrowLog : public ArrowLogBase {
   bool is_enabled_;
 
   static ArrowLogLevel severity_threshold_;
-  // In InitGoogleLogging, it simply keeps the pointer.
-  // We need to make sure the app name passed to InitGoogleLogging exist.
-  static std::unique_ptr<std::string> app_name_;
```

Comment:
No strong feelings about it, but why didn't you keep it a static class member?

Reply:
`glog` keeps `app_name_`'s pointer as a global variable... If we don't keep it as a static variable,  the string will be destructed. https://github.com/google/glog/blob/3267f3e1a8dd629922dbe28c09a5fa5e60de5a84/src/utilities.cc#L352

Reply:
You mean we don't call `ShutdownGoogleLoggingUtilities` sufficiently early?

Reply:
No. I mean `app_name_`'s pointer will still be used by `glog` after `InitGoogleLogging` is called. If we don't use static, `app_name_` will be destructed. The pointer kept by `glog` will be invalid.

Reply:
The variable was already static here. Hence my question.

Reply:
Since this variable is not used other function, @fsaintjacques suggested to move to local static member.